Fix for missing detector IDs in "resized" EventWorkspace.

SimonHeybrock opened the pull request:
This is an attempt to fix an issue introduced by #17360.

**To test:**

@peterfpeterson See if this works now
Fixes #17634.

No release notes, this issue was introduced only recently.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

peterfpeterson commented:
This was based off of `master` so it shouldn't be merged into `release-v3.8`. There is a fix for `SNSLiveData` already in #17633.

SimonHeybrock commented:
Fixed the base branch, and fixed similar to #17631 also for `TOPAZLiveEventDataListener`.

@rosswhitfield Could you please check if the `includeMonitors` flag is correct. Initially in this branch I had set it to `false`, which, I think, was the behavior of `EventWorkspace::padSpectra` (see https://github.com/mantidproject/mantid/pull/17360/files). However, in #17631 @peterfpeterson keeps it as the default (`true`), so I did that also here.

rosswhitfield commented:
@SimonHeybrock I think this is correct. I will go with what @peterfpeterson has done.

Anyway, I tested this and the TOPAZ live data work again. :shipit:
